client/state/jetpack-connect/reducer.js

```diff
@@ -123,19 +123,11 @@ export function jetpackSSO( state = {}, action ) {
 		case JETPACK_CONNECT_SSO_AUTHORIZE:
 			return Object.assign( state, { isAuthorizing: true } );
 		case JETPACK_CONNECT_SSO_VALIDATION_RECEIVE:
-			if ( action.error ) {
-				return Object. assign( state, { isValidating: false, validationError: action.error, nonceValid: false } );
-			}
-
-			return Object. assign( state, { isValidating: false, validationError: false, nonceValid: action.data.success } );
+			return Object. assign( state, { isValidating: false, validationError: action.error, nonceValid: action.success } );
 		case JETPACK_CONNECT_SSO_AUTHORIZATION_RECEIVE:
-			if ( action.error ) {
-				return Object. assign( state, { isAuthorizing: false, authorizationError: action.error, ssoUrl: false } );
-			}
-
-			return Object. assign( state, { isAuthorizing: false, authorizationError: false, ssoUrl: action.data.sso_url } );
-		case SERIALIZE:
+			return Object. assign( state, { isAuthorizing: false, authorizationError: action.error, ssoUrl: action.ssoUrl } );
 		case SERIALIZE:
+		case DESERIALIZE:
```

Comment:
While we're in this file, any chance we could fix https://github.com/Automattic/wp-calypso/blob/b6a8a750d826c3e2c9b47ac26e2e46992aef595c/client/state/jetpack-connect/reducer.js#L47 and https://github.com/Automattic/wp-calypso/blob/b6a8a750d826c3e2c9b47ac26e2e46992aef595c/client/state/jetpack-connect/reducer.js#L112

Steps would be
1. To double check if we want to persist these sub-trees
2. If it makes sense to persist, add a schema to guard against data drift
3. If it doesn't make sense, we'll want to return default state like we do here.

Also see: https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#data-persistence--2754-

If you have questions, feel free to ping me.